src/screens/ProfileScreen.tsx

```diff
@@ -1,186 +1,44 @@
-      title: '',
-      isDivider: false,
-    })
-
-  const handleRemoveUserAccount = useCallback(async () => {
-    try {
-      await removeUserAccount()
-      signOut()
-    } catch {
-      showErrorToast({
-        description: t('errors.something_went_wrong'),
-      })
-    }
-  }, [removeUserAccount, t])
-
-  const bottomSheet = bottomSheetComponentRenderFunction(
-    <Box px={4} pb={10}>
-      <Text.LgSemibold color="text.primary" pt={4} pb={2}>
-        {t('profile_screen.are_you_sure')}
-      </Text.LgSemibold>
-      <Text.SmRegular color="text.secondary" lineHeight="md">
-        {t('profile_screen.remove_account_desc')}
-      </Text.SmRegular>
-      <Row w="full" justifyContent="space-between" pt={8}>
-        <Button variant="SecondaryGray" flex={1} borderRadius={8} onPress={closeBottomSheet}>
-          {t('common.cancel')}
-        </Button>
-        <Spacer x={3} />
-        <Button
-          onPress={handleRemoveUserAccount}
-          variant="PrimaryDestructive"
-          flex={1}
-          borderRadius={8}
-          loading={isLoading}
-        >
-          {t('profile_screen.remove_account')}
-        </Button>
-      </Row>
-    </Box>,
-    {
-      name: 'delete-bin-line',
-      color: 'featured.icon.light.fg.error',
-      bgColor: 'bg.error.secondary',
-    }
-  )
+  const { isSubmitting, submit } = useUpdateProfileForm()
 
   useScreenOptions({
     title: t('navigation.screen_titles.profile'),
   })
 
-  const focusLastNameInput = useCallback(() => setFocus('lastName'), [setFocus])
-
   return (
     <Box p={4}>
-      <Text.LgBold color="text.primary">{t('profile_screen.profile')}</Text.LgBold>
-      <Spacer y={1} />
-      <Text.MdRegular color="text.secondary">
-        {t('profile_screen.update_your_details')}
-      </Text.MdRegular>
+      <ProfileHeader />
       <Spacer y={4} />
-      <Box borderColor="border.secondary" borderTopWidth={1} py={6}>
-        <Box
-          justifyContent="space-between"
-          flexDirection={isWeb && !shouldApplyMobileStyles ? 'row' : 'column'}
-          mb={isWeb ? 10 : 0}
-          maxW={800}
-        >
-          <Text.SmBold flex={1}>{t('form.labels.first_name')}</Text.SmBold>
-          <Box flex={isWeb ? 2 : 0}>
-            <ControlledField.Input
-              {...{ control, errors }}
-              autoCapitalize="none"
-              inputMode="text"
-              name="firstName"
-              onFocus={focusLastNameInput}
-              placeholder={t('form.placeholders.email')}
-              testID="emailInput"
-              {...(!isWeb && {
-                label: t('form.labels.first_name'),
-              })}
-            />
-          </Box>
-        </Box>
-        <Box
-          justifyContent="space-between"
-          flexDirection={isWeb && !shouldApplyMobileStyles ? 'row' : 'column'}
-          mb={isWeb ? 10 : 0}
-          maxW={800}
+      <ProfileDetailsForm />
+      <Row maxW={800} justifyContent="flex-end">
```

Comment:
Move buttons to `ProfileDetailsForm`

Reply:
Checked, thank you Mateusz, I'm sending throughout today new PRs